op-batcher/batcher/sync_actions.go

```diff
@@ -0,0 +1,144 @@
+package batcher
+
+import (
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-service/queue"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type channelStatuser interface {
+	isFullySubmitted() bool
+	isTimedOut() bool
+	LatestL2() eth.BlockID
+	MaxInclusionBlock() uint64
+}
+
+type syncActions struct {
+	clearState      *eth.BlockID
+	blocksToPrune   int
+	channelsToPrune int
+	blocksToLoad    [2]uint64 // the range [start,end] that should be loaded into the local state.
```

Comment:
I'm not a fan of using an array for a range. I'd find it clearer to just use either two fields like `loadBlocksStart` and `loadBlocksEnd` or make this a `*struct{start, end uint64}`. Wdyt?

Sorry for just remarking this now.

Reply:
[f588a00](https://github.com/ethereum-optimism/optimism/pull/13060/commits/f588a009c75168ef5819fd8868ebfb711db4b71c)